File: src/modules/InvoiceModule.js
/**
 * InvoiceModule.js
 * Responsibility: Format the data for a professional invoice.
 */
export const InvoiceModule = {
  generateInvoice(customerName, customerNumber, cartItems, subtotal, gst, discount, total) {
    return {
      invoiceNumber: `INV-${Date.now()}`,
      date: new Date().toLocaleDateString(),
      customerName,
      customerNumber,
      items: cartItems.map(item => ({
        name: item.name,
        price: item.price,
        quantity: item.quantity,
        total: item.price * item.quantity
      })),
      summary: {
        subtotal,
        gst,
        discount,
        total
      }
    };
  }
};

File: backend/InvoiceModule.cpp
#include "InvoiceModule.h"
#include <ctime>
#include <sstream>
#include <iomanip>

Invoice InvoiceModule::generateInvoice(const std::string& customerName, 
                                     const std::vector<CartItem>& cartItems, 
                                     double subtotal, double gst, double discount, double total) {
    time_t now = time(0);
    tm *ltm = localtime(&now);
    
    std::stringstream ss_num, ss_date;
    ss_num << "INV-" << now;
    ss_date << std::put_time(ltm, "%d/%m/%Y");

    return {
        ss_num.str(),
        ss_date.str(),
        customerName,
        cartItems,
        {subtotal, gst, discount, total}
    };
}
